cardano-api/src/Cardano/Api/Utils.hs

```diff
@@ -40,7 +39,7 @@ formatParsecError err =
     "expecting" "unexpected" "end of input"
     $ Parsec.errorMessages err
 
-runParsecParser :: Parsec.Parser a -> Text -> Aeson.Parser a
+runParsecParser :: MonadFail m => Parsec.Parser a -> Text -> m a
```

Comment:
Is there a specific reason why you have generalized this?

Reply:
The alternative would be to have `txInFromText :: Text -> Aeson.Parser TxIn` which then would only be possible to be used in an Aeson context, or not factor out `txInFromText`. The fact that nothing Aeson.Parser specific is used in `runParsecParser` and Aeson is only imported in Cardano.Api.Utils for this function kind of sealed the deal for me, but kan change it to one of the other alternatives if you want.

Reply:
I think its better to have the types concrete until we need to generalize them. So yeah, until there is a need for `txInFromText` in a different context lets leave it as `txInFromText :: Text -> Aeson.Parser TxIn`

Reply:
Done